cpp/src/io/parquet/page_decode.cuh

```diff
@@ -1247,7 +1267,13 @@ inline __device__ bool setupLocalPageInfo(page_state_s* const s,
           s->dict_val  = 0;
           if ((s->col.data_type & 7) == BOOLEAN) { s->dict_run = s->dict_size * 2 + 1; }
           break;
-        case Encoding::RLE: s->dict_run = 0; break;
+        case Encoding::RLE: {
+          // first 4 bytes are length of RLE data
+          int const len = (cur[0]) + (cur[1] << 8) + (cur[2] << 16) + (cur[3] << 24);
+          cur += 4;
+          if (cur + len > end) { s->error = 2; }
+          s->dict_run = 0;
```

Comment:
File this under "how did this ever work?".  What's the story here - how did RLE decoding work if we were starting 4 bytes off?

Reply:
> File this under "how did this ever work?". What's the story here - how did RLE decoding work if we were starting 4 bytes off?

I'm guessing you never ran into RLE encoded bool data before :wink:

Reply:
[Some light summer reading](https://issues.apache.org/jira/browse/PARQUET-2222) :wink: Near as I can tell booleans will only be encoded with RLE with V2 writers (although it seems arrow-rs might allow RLE w/ V1, but not by default). So it likely is true that this code has never been exercised.

Reply:
Looks like the encoder has RLE boolean support but it's disabled. If I enable it, it writes RLE encoded boolean data, but lacks the 4 bytes of length required by the Parquet spec. So I guess that answers the question of how did this ever work.

Once this PR and #13751 are merged I'll submit a PR to enable RLE encoding for booleans when writing V2 files.